### DISCUSSION TITLE
Arduino IDE

### DISCUSSION THREAD

**Discussion by tk5ep** [dl9rdz/rdz_ttgo_sonde#145] [Q&A]
Hi all,
I'm not sure if it's the right place to ask. If not, please correct me...

I tried to compile the sketch with my Arduino IDE 1.8.15, but i do get a lot of errors mainly related to the SX1278 and M10M20 libraries.
ex :
_C:\Users\Patrick\Documents\Arduino\libraries\SondeLib\SX1278FSK.cpp:32:23: error: 'RDZConfig {aka struct st_rdzconfig}' has no member named 'sx1278_ss'
  pinMode(sonde.config.sx1278_ss, OUTPUT);_

I use a ttgo_tbeam v1.1 and selected TBEAM in the board selector. Is this right ?

Can someone put me on the good track ?

Thanks, Patrick.

**Comment by dl9rdz** [comment]
I can only guess what is happening here.

My most likely guess is that you had an older version installed, and now did an update, and something went not as expected. See https://github.com/dl9rdz/rdz_ttgo_sonde/wiki/Arduino-IDE-installation, libraries "part 3". You will get the error you describe if you use the most recent SX1278FSK.cpp and a older version of Sonde.h.  

Make sure that all files from https://github.com/dl9rdz/rdz_ttgo_sonde/tree/devel/libraries/SondeLib are in your folder C:\Users\Patrick\Documents\Arduino\libraries\SondeLib\

If you are using git directly to update your copy of the source code and symbolic links from your Arduino library folder into the source code, then this should always update automatically (see https://www.howtogeek.com/howto/16226/complete-guide-to-symbolic-links-symlinks-on-windows-or-linux/ on how to use symbolic links in windows). If you instead use the "copy" approach for the libraries, make sure to copy all files each time you update to a new source code version.

**Reply by tk5ep** [reply]
Hi Hansi,
Thanks for the answer.

I had a wrong file somewhere, as after copying everything it worked.

I only had to deal with the display.cpp file.
The include path for the features.h file is highly dependent of the folder structure used. I had to modify the line : 

#include "../../RX_FSK/features.h"

so it works for me.

Everything compiles just fine and now i have to deal with the low sensitivity i've noticed. :)

Thanks and 73,
Patrick.

**Comment by dl9rdz** [comment]
Things have been simplified in recent version. There is no need to copy or link anything to some library folder.
Both ArudinoIDE and platformio should automatically pick up all cpp files from RX_FSK/src automatically
